sdk/eventhub/event-hubs/samples/eventProcessor.ts

```diff
@@ -1,43 +1,37 @@
 import {
   EventHubClient,
   ReceivedEventData,
-  EventPosition,
   delay,
   EventProcessor,
   PartitionContext,
   InMemoryPartitionManager,
-  CheckpointManager
+  PartitionProcessor,
+  CloseReason
 } from "@azure/event-hubs";
 
-class SimplePartitionProcessor {
-  private _context: PartitionContext;
-  private _checkpointManager: CheckpointManager;
-  constructor(context: PartitionContext, checkpointManager: CheckpointManager) {
-    this._context = context;
-    this._checkpointManager = checkpointManager;
-  }
-  async processEvents(events: ReceivedEventData[]) {
-    if(events.length === 0){
+class SamplePartitionProcessor extends PartitionProcessor {
+  private _messageCount = 0;
+
+  async processEvents(events: ReceivedEventData[], partitionContext: PartitionContext) {
+    if (events.length === 0) {
       return;
     }
     for (const event of events) {
       console.log(
-        "Received event: '%s' from partition: '%s' and consumer group: '%s'",
-        event.body,
-        this._context.partitionId,
-        this._context.consumerGroupName
+        `Received event: '${event.body}' from partition: '${partitionContext.partitionId}' and consumer group: '${partitionContext.consumerGroupName}'`,
       );
       try {
         // checkpoint using the last event in the batch
-        await this._checkpointManager.updateCheckpoint(events[events.length - 1]);
+        await partitionContext.updateCheckpoint(events[events.length - 1]);
```

Comment:
Why are we checkpointing inside the for loop?

Reply:
Good point. I only looked at changing the API to use the new version, missed the fact that it was in the for loop this whole time. I'll update it.